libsolidity/formal/SMTEncoder.cpp

```diff
@@ -779,7 +779,16 @@ void SMTEncoder::visitTypeConversion(FunctionCall const& _funCall)
 	auto argument = _funCall.arguments().front();
 	unsigned argSize = argument->annotation().type->storageBytes();
 	unsigned castSize = _funCall.annotation().type->storageBytes();
-	if (argSize == castSize)
+	// Allow same-size casts as well as literal arguments.
+	// TODO: support StringLiteral
+	if (auto category = argument->annotation().type->category(); category == Type::Category::RationalNumber)
```

Comment:
Perhaps we should restrict this to AddressType and IntegerType initially?

Reply:
I'm not convinced by the way the `fixed()` test case is processed.

Reply:
As long as fixed is not implemented properly (for now it's abstracted as uint256) I don't think it makes sense to separate it.